File: vonage/python/generate-signature/generate-signature.py
import os
import sys
from os.path import dirname, join
from dotenv import load_dotenv
from vonage_jwt.jwt import JwtClient

dotenv_path = join(dirname(__file__), '../../.env')
load_dotenv(dotenv_path)

application_id = os.environ.get('VONAGE_APPLICATION_ID')
application_private_key = os.environ.get('VONAGE_APPLICATION_PRIVATE_KEY')

try:
    client = JwtClient(application_id, application_private_key);
    print(client.generate_application_jwt().decode())
except Exception as e:
    print('Exception %s: %s' % (type(e), e))
    sys.exit(1)

sys.exit

File: vonage/php/generate-signature/generate-signature.php
<?php
require_once sprintf('%s/vendor/autoload.php', dirname(__DIR__, 1));

use Vonage\JWT\TokenGenerator;

$dotenv = Dotenv\Dotenv::createImmutable(dirname(__DIR__, 2));
$dotenv->load();

$application_id = $_ENV['VONAGE_APPLICATION_ID'];
$application_private_key = $_ENV['VONAGE_APPLICATION_PRIVATE_KEY'];

try {
    $generator = new TokenGenerator($application_id, $application_private_key);
    echo $generator->generate(), "\n";
} catch (Exception $e) {
    echo sprintf('Exception %s: %s at line %d in %s',
        get_class($e), $e->getMessage(), $e->getLine(), $e->getFile()), "\n";
    exit(1);
}

exit;
